addons/binding/org.openhab.binding.smappee/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,180 @@
+                <context>password</context>
+                <advanced>false</advanced>
+            </parameter>
+            
+            <parameter name="service_location_name" type="text" required="true">
+                <label>Service location name</label>
+                <description>The name of your Smappee installation</description>
+                <advanced>false</advanced>
+            </parameter>
+            
+            <parameter name="poll_time" type="integer" required="true" min="1" max="1440">
+                <label>Polling time</label>
+                <description>How often (in minutes) does the smappee needs to be checked ?</description>
+                <default>5</default>
+                <advanced>false</advanced>
+            </parameter>
+            
+        </config-description>                
+    </bridge-type>
+
+    <channel-type id="smappee-consumption-channel">
+        <item-type>Number</item-type>
+        <label>Energy Consumption</label>
+        <description>The amount of energy that is being consumed</description>
+    </channel-type>
+    
+    <channel-type id="smappee-solar-channel">
+        <item-type>Number</item-type>
+        <label>Solar Energy</label>
+        <description>The amount of energy that is being generated by your solar power panels</description>
+    </channel-type>
+        
+    <channel-type id="smappee-alwayson-channel">
+        <item-type>Number</item-type>
+        <label>AlwaysOn</label>
+        <description>The amount of energy that is always consumed</description>
+    </channel-type>
+
+    <thing-type id="smappee-appliance">
+        <supported-bridge-type-refs>
+            <bridge-type-ref id="smappee" />
+        </supported-bridge-type-refs>
+        
+        <label>Smappee Appliance</label>
+        <description>Detected Smappee Appliance</description>
+
+        <channels>
+            <channel id="appliance-power" typeId="smappee-appliance-power"/>
+            <channel id="appliance-lastupdate" typeId="smappee-appliance-lastupdate"/>
+        </channels>
+        
+        <config-description>
+            <parameter name="id" type="text">
+                <label>ID</label>
+                <description>Unique identifier for the appliance</description>
+                <required>true</required>
+            </parameter>
+            
+            <parameter name="type" type="text">
+                <label>Type</label>
+                <description>The appliance type</description>
+                <required>true</required>
+            </parameter>           
+        </config-description>
+    </thing-type>
+    
+    <channel-type id="smappee-appliance-power">
+        <item-type>Number</item-type>
+        <label>Name</label>
+        <description>The current power consumption of the appliance</description>
+    </channel-type>
+    
+    <channel-type id="smappee-appliance-lastupdate">

```

Comment:
no, this is usefull, appliances can go off the grid, this can be a physical reason (somebody pulled the plug out) or the api itself, I found out that sometimes no data is collected anymore for a certain appliance, maybe because the energy consumption pattern changed. this channel helps detecting this.